contracts/staking/ClusterRewards.sol

```diff
@@ -251,10 +253,24 @@ contract ClusterRewards is
 
     function _processReceiverTickets(address _receiver, uint256 _epoch, address[] memory _selectedClusters, uint16[] memory _tickets, uint256 _totalNetworkRewardsPerEpoch, uint256 _epochTotalStake, uint256 _epochReceiverStake) internal {
         require(!_isTicketsIssued(_receiver, _epoch), "CRW:IPRT-Tickets already issued");
-
         unchecked {
             require(_selectedClusters.length <= _tickets.length + 1, "CRW:IPRT-Tickets length not matching selected clusters");
             uint256 _rewardShare = _totalNetworkRewardsPerEpoch * _epochReceiverStake / _epochTotalStake;
+            uint256 _receiverRewardPerEpoch = receiverRewardPerEpoch[_receiver];
+            
+            // if(receiverBalance[_receiver] > _receiverRewardPerEpoch){
+            //     _rewardShare += _receiverRewardPerEpoch;
+            //     receiverBalance[_receiver] -= _receiverRewardPerEpoch;
+            // }
+
+            // using least number of variables to avoid stack overflow
+            uint256 _temp = receiverBalance[_receiver];
+            _temp = _temp > _receiverRewardPerEpoch ? _receiverRewardPerEpoch: _temp;
+            if(_temp != 0){
+                receiverBalance[_receiver] -= _temp;
+                _rewardShare += _temp;
```

Comment:
Calculate rewardShare outside this function, as this part of code is just dependent on receiver address which is known before the for loop which calls this function. This ensures that the storage accesses  are minimal if  this is done outside the for loop instead.

Reply:
Done